ktor-server/jvm/src/io/ktor/server/application/plugins/api/ServerPlugin.kt

```diff
@@ -12,31 +12,56 @@ import io.ktor.server.request.*
 import io.ktor.server.response.*
 import io.ktor.util.*
 import io.ktor.util.pipeline.*
-import kotlin.random.Random
+import kotlin.random.*
 
 /**
  * A plugin that embeds into the HTTP pipeline and extends Ktor functionality.
  **/
-public abstract class ServerPlugin<Configuration : Any> private constructor(
-    pluginFactory: ServerPluginFactory<Configuration>
+public sealed class ServerPlugin<Configuration : Any> private constructor(
```

Comment:
What if a user wants to extend this class as well?

Reply:
Is there a reason for it? I prefer to forbid inheritance until use cases are clear, because it's something that is hard to take away or change when needed.